llvm/lib/Analysis/IVDescriptors.cpp

```diff
@@ -1063,7 +1068,7 @@ unsigned RecurrenceDescriptor::getOpcode(RecurKind Kind) {
 SmallVector<Instruction *, 4>
 RecurrenceDescriptor::getReductionOpChain(PHINode *Phi, Loop *L) const {
   SmallVector<Instruction *, 4> ReductionOperations;
-  unsigned RedOp = getOpcode();
+  bool IsNonArithmetic = !isArithmeticRecurrenceKind(Kind);
```

Comment:
I think I'd prefer this to be:

```
  bool IsComparison = Instruction::ICmp || RedOp == Instruction::FCmp;
```

or something like that. It's more precise and we don't have to worry about updating `isArithmeticRecurrenceKind` every time we add a new recurrence kind. It also explicitly excludes RecurKind::None.